File: c/2/2312.c
#include <stdio.h>
#include <math.h>

void tick()
{
    int number;
    scanf("%d", &number);

    int limit = ceil(sqrt(number));
    for (int i = 2; i <= limit; i++)
    {
        int count = 0;
        while (number % i == 0)
        {
            number /= i;
            count++;
        }

        if (count > 0)
        {
            printf("%d %d\n", i, count);
        }
    }

    if (number != 1)
    {
        printf("%d 1\n", number);
    }
}

int main(void)
{
    int count;
    scanf("%d", &count);

    for (int i = 0; i < count; i++)
    {
        tick();
    }

    return 0;
}

File: python/2/2312.py
from sys import stdin
from math import sqrt, ceil

input = stdin.readline


def tick():
    number = int(input())

    for i in range(2, ceil(sqrt(number) + 1)):
        count = 0
        while number % i == 0:
            number //= i
            count += 1

        if count > 0:
            print(i, count)

    if number != 1:
        print(number, 1)


def main():
    for _ in range(int(input())):
        tick()


if __name__ == '__main__':
    main()
